pkg/providers/amifamily/windows.go

```diff
@@ -0,0 +1,78 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package amifamily
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/karpenter/pkg/providers/amifamily/bootstrap"
+
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+)
+
+type Windows struct {
+	DefaultFamily
+	*Options
+	template *v1alpha1.AWSNodeTemplate
+}
+
+// SSMAlias returns the AMI Alias to query SSM
+func (w Windows) SSMAlias(version string, _ *cloudprovider.InstanceType) string {
+	windowsVersion := getValueFromAnnotation(w.template.Annotations, v1alpha1.AnnoInstanceWindowsVersion, v1alpha1.Windows2019)
+	windowsVariant := getValueFromAnnotation(w.template.Annotations, v1alpha1.AnnoInstanceWindowsVariant, v1alpha1.WindowsCore)
+	return fmt.Sprintf("/aws/service/ami-windows-latest/Windows_Server-%s-English-%s-EKS_Optimized-%s/image_id", windowsVersion, windowsVariant, version)
+}
+
+func getValueFromAnnotation(annotations map[string]string, key string, defaultValue string) string {
+	result := annotations[key]
+	if result == "" {
+		result = defaultValue
+	}
+	return result
+}
+
+// UserData returns the default userdata script for the AMI Family
+func (w Windows) UserData(kubeletConfig *v1alpha5.KubeletConfiguration, taints []v1.Taint, labels map[string]string, caBundle *string, _ []*cloudprovider.InstanceType, customUserData *string) bootstrap.Bootstrapper {
+	return bootstrap.Windows{
+		Options: bootstrap.Options{
+			ClusterName:     w.Options.ClusterName,
+			ClusterEndpoint: w.Options.ClusterEndpoint,
+			KubeletConfig:   kubeletConfig,
+			Taints:          taints,
+			Labels:          labels,
+			CABundle:        caBundle,
+			CustomUserData:  customUserData,
+		},
+	}
+}
+
+// DefaultBlockDeviceMappings returns the default block device mappings for the AMI Family
+func (w Windows) DefaultBlockDeviceMappings() []*v1alpha1.BlockDeviceMapping {
+	return []*v1alpha1.BlockDeviceMapping{{
+		DeviceName: w.EphemeralBlockDevice(),
+		EBS:        &DefaultEBS,
```

Comment:
By default on EKS Optimized Windows AMIs, the EBS configuration is-
```
{
  "blockDeviceMappings": [
    {
      "DeviceName": "/dev/sda1",
      "Ebs": {
        "DeleteOnTermination": true,
        "Encrypted": false,
        "VolumeSize": 50,
        "VolumeType": "gp2"
      }
    }
  ]
}
```

Should we change this to make them consistent?

Reply:
50G is required. Let's use a similar config as other AmiFamily. for example
https://github.com/aws/karpenter/blob/07c1ebf5942e4630659a4de25f58b7b6244201cc/pkg/providers/amifamily/bottlerocket.go#L73

Reply:
@rawahars Is there any particular reason that the volumeSize is larger for windows in the default case? Would it be fine to default to `20Gi` and `gpt3`?

Reply:
So Windows EKS AMIs do have cached container images which take substantial space. Also, Windows base images are larger than Linux ones. For example- `mcr.microsoft.com/windows:ltsc2019` is around 11.2 Gi when uncompressed.

Therefore, we keep the default size to `50Gi`. I think we can default the `VolumeType` to `gpt3`.

Reply:
> @rawahars Is there any particular reason that the volumeSize is larger for windows in the default case? Would it be fine to default to `20Gi` and `gpt3`?

The Windows AMI root block is 50Gi. So 20Gi is not an option to launch the instance and will be rejected by ec2.